### PR TITLE
Update typescript definitions

### PR DESCRIPTION
**Pull request by HenriBeck**
__What would you like to add/fix?__

Adjusts the typescript definitions

__Corresponding issue:__
#1215
#1210
#1203
#1199

### CODE REVIEW

**Comment by kof** [comment]
I would suggest when it's about ts types, you just go ahead and merge it, since I can't tell much if they are ok.

**Comment by kof** [comment]
Don't forget the changelog

**Comment by kof** [comment]
going to merge it now for the next release

**Comment by kof** [comment]
Merged master, there is an error, it seems obvious but I am not sure what was the plan, what are you testing there, can you please have a look?

File: packages/jss/tests/jss-tests.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import {create as createJSS, createGenerateId, SheetsRegistry, default as shared
 const jss = createJSS().setup({createGenerateId})
 jss.use({}, {}) // $ExpectType JSS
 
-const styleSheet = jss.createStyleSheet<string>(
+const styleSheet = jss.createStyleSheet<string, {color: string}>(
```
Comment (kof) [inline packages/jss/tests/jss-tests.ts]: this is failing

**Comment by kof** [comment]
We should write a summary of all tha in changelog

**Comment by HenriBeck** [comment]
> We should write a summary of all tha in changelog

Done

**Comment by kof** [comment]
Any of the PR can be also closed because already fixed here?

**Comment by kof** [comment]
released in 10.0.3

**Pull request merged**